app/services/idv/engine/events.rb

```diff
@@ -0,0 +1,277 @@
+    :idv_user_consented_to_share_pii,
+    :idv_user_entered_password,
+    :idv_user_entered_ssn,
+    :idv_user_started,
+    :idv_user_updated_mailing_address,
+    :idv_user_verified_their_info,
+  ].freeze
+
+  AuthUserChangedPasswordParams = Struct.new(
+    :password,
+    keyword_init: true,
+  )
+
+  IdvDocumentsSubmittedToAcuantParams = Struct.new(
+    :request_success,
+    :request_timed_out,
+    :response_status,
+    :result,
+    :billed,
+    :processed_alerts,
+    :alert_failure_count,
+    :address_line2_present,
+    keyword_init: true,
+  )
+
+  IdvDocumentsSubmittedToTrueidParams = Struct.new(
+    :request_success,
+    :request_timed_out,
+    :response_status,
+    :result,
+    :billed,
+    :processed_alerts,
+    :alert_failure_count,
+    :address_line2_present,
+    keyword_init: true,
+  )
+
+  IdvFraudThreatmetrixCheckCompletedParams = Struct.new(
+    :request_success,
+    :request_timed_out,
+    :response_status,
+    :threatmetrix_session_id,
+    :threatmetrix_review_status,
+    keyword_init: true,
+  )
+
+  IdvFraudThreatmetrixCheckInitiatedParams = Struct.new(
+    :threatmetrix_session_id,
+    keyword_init: true,
+  )
+
+  IdvUserEnteredPasswordParams = Struct.new(
+    :password,
+    keyword_init: true,
+  )
+
+  IdvUserEnteredSsnParams = Struct.new(
+    :ssn,
+    keyword_init: true,
+  )
+
+  IdvUserUpdatedMailingAddressParams = Struct.new(
+    :address,
+    keyword_init: true,
+  )
+
+  # The user has changed their password.
+  # @param [Object] params
+  # @return [AuthUserChangedPasswordParams]
+  def auth_user_changed_password(params)
+    raise 'params cannot be nil' if params.nil?
+
+    if params.is_a?(Hash)
+      params = AuthUserChangedPasswordParams.new(params)
+    end
+
+    handle_event :auth_user_changed_password, params
+    params
+  end
+
+  # The user has reset their password and will not be able to access their IDV
+  # PII until providing their personal key.
+  # @return [nil]
+  def auth_user_reset_password
+    handle_event :auth_user_reset_password
+    nil
+  end
+
+  # The user's address was submitted to LexisNexis InstantVerify.
+  # @return [nil]
+  def idv_address_submitted_to_instantverify
+    handle_event :idv_address_submitted_to_instantverify
+    nil
+  end
+
+  # The user has uploaded identity documents, and Login.gov has submitted them
+  # to Acuant for  processing.
+  # @param [Object] params
+  # @return [IdvDocumentsSubmittedToAcuantParams]
+  def idv_documents_submitted_to_acuant(params)
+    raise 'params cannot be nil' if params.nil?
+
+    if params.is_a?(Hash)
+      params = IdvDocumentsSubmittedToAcuantParams.new(params)
+    end
+
+    handle_event :idv_documents_submitted_to_acuant, params
+    params
+  end
+
+  # The user has uploaded identity documents, and Login.gov has submitted them
+  # to LexisNexis  TrueID for processing.
+  # @param [Object] params
+  # @return [IdvDocumentsSubmittedToTrueidParams]
```

Comment:
I think this is a good example of why using `String#camelize` can be bad, because `Trueid` capitalization is a little off. I think it would be good case for allowing an explicit (but optional) class name override in the `.yml`